tests/testthat/test-write_data.R

```diff
@@ -0,0 +1,39 @@
+dummy_data <- tibble(
+  chr = sample(LETTERS, 100, replace = TRUE),
+  int = sample(0:9, 100, replace = TRUE),
+  num = runif(100)
+)
+
+test_that("write_sav creates a file with the correct permissions", {
+  temp_path <- tempfile()
+  write_sav(dummy_data, temp_path)
+
+  file_info <- fs::file_info(temp_path)
+  read_data <- haven::read_sav(temp_path)
+
+  # Check data is the same
+  expect_equal(haven::zap_formats(read_data), dummy_data)
+
+  # Check the permissions are as expected
+  expect_match(
+    as.character(file_info$permissions),
+    "rw\\-rw\\-\\-\\-\\-"
+  )
+})
```

Comment:
sav test failed for me? unsure if you got this when ran. 

`── Error (Line 3): write_sav creates a file with the correct permissions ─────────────────────
Error: Expected single logical value
Backtrace:
 1. global write_sav(dummy_data, temp_path)
 2. haven::write_sav(data = data, path = path, compress = compress)
 3. haven:::write_sav_(...)`

Reply:
I don't get this error here, the tests seem to be working okay for me

Reply:
Can you confirm you still get an error, Catherine? Is it possible that you're using an old version of haven? I'm using the `compress` argument `"zsav"` which is only valid from [version 2.5.0](https://haven.tidyverse.org/news/index.html#haven-250), `TRUE` and `FALSE` still work for backwards compatibility but using `"zsav"` is clearer.

I suspect your error is that the old version expects a logical so doesn't like the `"zsav"`  bit.

Reply:
updated `haven` and test works now :)